Log cache state to structured logs

oulgen opened the pull request:
https://interncache-all.fbcdn.net/manifold/tlparse_reports/tree/logs/.tmpRm4MaD/0_0_0/fx_graph_cache_hash_4.json

Differential Revision: D59795574


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

jamesjwu commented on torch/_inductor/codecache.py:
```diff
@@ -574,7 +574,7 @@ def get_hash(cls, obj: Any) -> str:
 
     @classmethod
-    def debug_str(cls, inp: Any) -> str:
+    def debug_lines(cls, inp: Any) -> List[str]:
```
Minor, unrelated point: when I ran strobelight on this, this one function is 0.7% of compilation time for FXGraphCache, and another 0.7% for AOTAutogradCache. I was going to fix this by only calling the function when log level was set to debug, but I realize now that since we want these logs for tlparse this is just always going to be a waste of time. 
https://fburl.com/scuba/pyperf_experimental/on_demand/x7pzb7z7

We might want to consider storing debug_lines in the cache entry so that on a cache hit we don't need to recalculate it? That said, maybe 1.5% warm start time is pretty small and won't matter

ezyang replied:
Yes, please plan for things that go into structured trace to be hit all the time. To be honest, we haven't really interrogated how much compile time slow down we are willing to eat with TORCH_TRACE, a random number I'll pull out of my ass is 10%. We don't instrument the costs here but it would be good to do so so we can evaluate how we're doing.

oulgen replied:
> 0.7% of compilation time

I assume the bigger the model is less percentage of compile time this number will be.

oulgen committed:
Log cache state to structured logs (#130845)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/130845

Test Plan: https://interncache-all.fbcdn.net/manifold/tlparse_reports/tree/logs/.tmpRm4MaD/0_0_0/fx_graph_cache_hash_4.json

Reviewed By: jamesjwu

Differential Revision: D59795574
